packages/bbui/src/Form/Core/DatePicker.svelte

```diff
@@ -64,9 +65,8 @@
     if (newValue) {
       newValue = newValue.toISOString()
     }
-
     // If time only set date component to 2000-01-01
-    if (timeOnly) {
+    else if (timeOnly) {
```

Comment:
What was the reason for adding this `else`? Adding that in will prevent the `timeOnly` logic from ever running if there's an existing value.